src/assets/custom/js/websiteHeader.test.js

```diff
@@ -1,120 +1,139 @@
+          expect(header.classList).not.toContain('website-header--revealed');
         });
       });
 
-      describe("When we then scroll up the page by at least three pixels", () => {
+      describe('When we then scroll up the page by at least three pixels', () => {
         beforeEach(() => {
           simulateScrollingToY(1000 - 3);
         });
 
-        it("we reveal the header positioned at the top of the viewport", () => {
-          expect(header.classList).toContain("website-header");
-          expect(header.classList).toContain("website-header--revealed");
+        it('we reveal the header positioned at the top of the viewport', () => {
+          expect(header.classList).toContain('website-header');
+          expect(header.classList).toContain('website-header--revealed');
           expect(header.classList.length).toBe(2);
         });
 
-        describe("When we then scroll close to the top of the page", () => {
-          beforeEach(() => {
-            simulateScrollingToY(1);
+        describe('When we then open a submenu', () => {
+          beforeAll(() => {
+            header.classList.add(HEADER_HAS_OPEN_SUBMENU_CLASS);
           });
-
-          it("we let the header return to it's natural position", () => {
-            expect(header.classList).toContain("website-header");
-            expect(header.classList.length).toBe(1);
+          afterAll(() => {
+            header.classList.remove(HEADER_HAS_OPEN_SUBMENU_CLASS);
           });
-        });
+          describe('And we scroll down the page by three pixels', () => {
+            beforeEach(() => {
+              simulateScrollingToY(1000);
+            });
 
-        describe("When we then scroll down the page by a couple of pixels", () => {
-          beforeEach(() => {
-            simulateScrollingToY(1000 - 1);
-          });
+            it('The submenu closes', () => {
+              expect(header.classList).not.toContain(
+                HEADER_HAS_OPEN_SUBMENU_CLASS
+              );
+            });
 
-          it("we don't immediately hide the header", () => {
-            expect(header.classList).toContain("website-header");
-            expect(header.classList).toContain("website-header--revealed");
-            expect(header.classList.length).toBe(2);
+            it('The header is hidden', () => {
+              expect(header.classList).not.toContain(
+                'website-header--revealed'
+              );
+            });
           });
         });
 
-        describe("When a submenu is open", () => {
-          beforeAll(() => {
-            header.classList.add(HEADER_HAS_OPEN_SUBMENU_CLASS);
-          });
-
-          describe("And we scroll down the page by three pixels", () => {
+        describe('when a submenu is not open', () => {
+          describe('And we scroll down the page by three pixels', () => {
             beforeEach(() => {
-              simulateScrollingToY(1000);
+              simulateScrollingToY(1000 + 100);
```

Comment:
The description talks about "three pixels" because we were testing the boundaries. Now this arrangement is introducing much bigger steps. Careful to keep it readable and understandable